readthedocs/core/views/hooks.py

```diff
@@ -100,7 +101,17 @@ def sync_versions(project):
         if not version:
             log.info('Unable to sync from %s version', version_identifier)
             return None
-        sync_repository_task.delay(version.pk)
+        task = chain(
+            sync_repository_task.signature(
+                args=(version.pk,),
+                immutable=True,
+            ),
+            clean_build_task.signature(
+                args=(version.pk,),
+                immutable=True,
+            ),
+        )
+        task.delay()
```

Comment:
This line will trigger a chain of `sync_repository_task` and `clean_build_task`. How do we guarantee that these tasks are going to be executed both in the same server?